Mathlib/Analysis/Complex/UpperHalfPlane/Manifold.lean

```diff
@@ -34,4 +35,39 @@ theorem mdifferentiable_coe : MDifferentiable 𝓘(ℂ) 𝓘(ℂ) ((↑) : ℍ 
   smooth_coe.mdifferentiable
 #align upper_half_plane.mdifferentiable_coe UpperHalfPlane.mdifferentiable_coe
 
+local notation "↑ₕ" f => f ∘ (PartialHomeomorph.symm
+          (OpenEmbedding.toPartialHomeomorph UpperHalfPlane.coe openEmbedding_coe))
+
+/--This shows that being MDifferentiable as a map `ℍ → ℂ` is equivalent to being
+differentiable on `{z : ℂ | 0 < z.im}` after arbitrarily extending to a function on all of `ℂ`.-/
```

Comment:
I'm surprised this is so painful. The comparable argument for Jacobi theta functions (in JacobiTheta.Manifold) is literally a one-liner, why doesn't the same work here?

Reply:
Ah I didn't know about `DifferentiableAt.mdifferentiableAt` which is the key here, this way I don't need that other ugly lemma which I was wondering about. I'll incorporate this and try again.